Too much import

outofculture opened the pull request:
This PR is a little more dangerous. Double-check me on c354e28ca and 5044115f5 especially.

outofculture committed:
docker for qt5 under python2

outofculture committed:
most of the import-stars removed

outofculture committed:
local import to prevent circular import

outofculture committed:
Merge branch 'toward-python3' into so-many-versions

outofculture committed:
utils that just pass along pyqtgraph can go away

outofculture committed:
Merge branch 'so-many-versions' into too-much-import

outofculture committed:
tracking down more import errors

outofculture committed:
more import fixes

outofculture committed:
integers needed for index slicing, so make sure it looks int-y

outofculture committed:
get the errno off the exception

campagnola commented on acq4/devices/Camera/Camera.py:
```diff
@@ -53,9 +52,9 @@ class Camera(DAQGeneric, OptomechDevice):
 
     sigCameraStopped = Qt.Signal()
     sigCameraStarted = Qt.Signal()
-    sigShowMessage = Qt.Signal(object)  # (string message)
```
These should stay as `object`

campagnola commented on acq4/devices/Camera/Camera.py:
```diff
@@ -770,8 +768,8 @@ def frameTimes(self):
         
 class AcquireThread(Thread):
     
-    sigNewFrame = Qt.Signal(object)
```
and here

campagnola commented on acq4/devices/NiDAQ/resample_test.py:
```diff
@@ -1,30 +1,33 @@
-data[dlen*0.4]+= 1000
-data += sin(xVals*40677*2.0*pi)*4.
+data += np.random.normal(size=dlen) + 20.
+data[round(dlen*0.102):round(dlen*0.3)] += 20
```
Shouldn't these be `int` instead of `round` ?

outofculture replied:
Good question! I didn't actually look up what the previous behavior would have been, but the math worked out that all of these `dlen * blah` expressions were supposed to be whole numbers, so if `int` and `round` had different answers in these cases due to floating point noise, `round` would almost certainly be the intended result. It may well be the case that the previous behavior was actually `int`, in which case this change probably fixes an unnoticed bug.

campagnola replied:
Not just FP noise; `round(x)` and `int(x)` give different answers whenever `x % 1 > 0.5`. This particular example should work either way, but the previous numpy behavior (now deprecated) was to coerce float indices using `int`. Not a bug; it was the expected behavior.

outofculture replied:
Mm, yes, I did check first that none of the rounded numbers were supposed to have decimal parts.

campagnola commented on acq4/devices/PVCam/PVCam.py:
```diff
@@ -99,7 +99,7 @@ def noFrameWarning(self, time):
         # camera can cause it to fail to return frames, even if there is no data being sent to/from the serial devices.
         # 
-        print("Camera acquisition thread has been waiting %02f sec but no new frames have arrived; shutting down." % diff)
+        print("Camera acquisition thread has been waiting %02f sec but no new frames have arrived; shutting down." % np.diff)
```
`time` instead of `np.diff`?

outofculture committed:
Revert accidental commit of generated cache file

campagnola commented on acq4/modules/Patch/PatchWindow.py:
```diff
@@ -2,29 +2,30 @@
 
 class PatchWindow(Qt.QMainWindow):
     
-    sigWindowClosed = Qt.Signal(object)
```
`object`

campagnola commented on acq4/modules/Patch/PatchWindow.py:
```diff
@@ -76,7 +77,7 @@ def __init__(self, dm, config):
         #self.statusBar().addPermanentWidget(self.logBtn)
         self.setStatusBar(StatusBar())
 
-        self.stateFile = os.path.join('modules', self.clampName + '_ui.cfg')
```
A bunch more in here -- np.os, np.sys, np.np, np.str

outofculture replied:
Mm, fixed. And some added `np.int` and `np.float` additions that can be removed.

campagnola commented on acq4/util/clibrary/CParser.py:
```diff
@@ -347,9 +348,9 @@ def removeComments(self, file):
         """Remove all comments from file. (operates in memory; does not alter the original files)"""
         self.assertPyparsing()
         text = self.files[file]
-        cplusplusLineComment = Literal("//") + restOfLine
```
These changes seem to make readability suffer in some already dense code.. thoughts? 
Maybe `from pyparsing import (a, bunch, of, specific, stuff)` ?

outofculture replied:
Yeah, that's an additional 4 keystrokes per import to change; I'll go through and hit the noisier modules.

outofculture replied:
I did it for pyparsing and ctypes in a few places. I'll keep an eye out for more opportunities to shorten code this way when it's clear what the libraries in play are.

outofculture committed:
fixes from review

remove np. from libraries it has nothing to do with
  as well as np.np.
pyparsing imports converted to explicitly named identifiers

outofculture committed:
np.int and np.float are just builtins

outofculture committed:
ctypes is used on lots of long, noisey lines, so make imports explicit

outofculture committed:
ctypes is used on lots of long, noisey lines, so make imports explicit

outofculture committed:
remove platform-specific functype imports

outofculture committed:
remove platform-specific ctypes-dll import
